### PR TITLE
Add option for skipping changes of some attributes

### PR DESCRIPTION
**Pull request by haylen**
This PR allows us to skip some attributes from being stored in the `item_changes` column. 

Steps to achieve this: 
1. Go to your model that supports versions
2. Define `paper_trail_skip` function which returns a list of attributes you want to skip

```elixir
def paper_trail_skip, do: [:twitter, :facebook]
```
**NOTE:** But there is one issue I can't solve. For some reasons tests randomly fail when I comment out some of the tests in the [PaperTrailTest.SkippedAttributesDetector](https://github.com/haylen/paper_trail/blob/add-skip-feature/test/paper_trail/skipped_attributes_detector_test.exs). Would really appreciate if someone could take a look.

**UPD:** Looks like an issue is related to my local settings as test suit passes on my colleague's machine, as tests that are failing related to postgres

### CODE REVIEW

**Comment by izelnakri** [comment]
Thanks for your contribution! I'll have a look at this whenever I have time.

**Comment by izelnakri** [comment]
Normally I've been against implementing this feature, but then I checked your implementation and realized how clean and simple it is, well done! Although this feature will make version reversion implementations more complex Im willing to merge this request!

**Comment by izelnakri** [comment]
Can you add documentation to this feature on README.md?

File: lib/paper_trail/skipped_attributes_detector.ex [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  Gets list of skipped attributes for a module
+  """
+  @spec call(module :: atom()) :: List.t()
+  def call(module) when is_atom(module) do
+    case :erlang.function_exported(module, :paper_trail_skip, 0) do
+      true ->
+        attrs = module.paper_trail_skip
+
+        case is_list(attrs) do
+          true ->
```
Comment (izelnakri) [inline lib/paper_trail/skipped_attributes_detector.ex]: I prefer true -> attrs

File: lib/paper_trail/skipped_attributes_detector.ex [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            attrs
+          _ ->
+            raise ArgumentError
```
Comment (izelnakri) [inline lib/paper_trail/skipped_attributes_detector.ex]: _ -> raise ArgumentError

**Comment by izelnakri** [comment]
please also run the `mix format` command. Some tests are breaking locally in my computer too. I'm investigating it now.

**Comment by izelnakri** [comment]
Ive rebased from master and upgraded my elixir environment, seems like the problem stems from the test migration files, they dont create the relevant Postgres sequences. I'll dip deeper into in when I find time, meanwhile feel free to take a stab at it!

**Comment by haylen** [comment]
@izelnakri thanks for a feedback. I updated Readme and reformatted the code. Also was investigating further an issue with failing tests but no luck so far. The only thing I see is that it's not related to Ecto 3.0, as tests still fail in the [branch](https://github.com/haylen/paper_trail/tree/skip-feature-with-0.8.0) with Ecto 2.1. Weird thing is that they pass when I comment out 8-30 lines in the `PaperTrailTest.SkippedAttributesDetector`. Those tests don't even need repo, as they don't access database, but somehow influence whole test suit and makes it fail with such errors:
```
  1) test creates versions with models that have a UUID primary key (PaperTrailTest.UUIDTest)
     test/paper_trail/uuid_test.exs:27
     ** (Postgrex.Error) ERROR 42P01 (undefined_table) relation "products_id_seq" does not exist

         query: select last_value FROM products_id_seq
     code: |> PaperTrail.insert!()
     stacktrace:
       (ecto_sql) lib/ecto/adapters/sql.ex:590: Ecto.Adapters.SQL.raise_sql_call_error/1
       (paper_trail) lib/paper_trail.ex:351: PaperTrail.get_sequence_id/1
       (paper_trail) lib/paper_trail.ex:104: anonymous fn/3 in PaperTrail.insert!/2
       (ecto_sql) lib/ecto/adapters/sql.ex:784: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
       (db_connection) lib/db_connection.ex:1341: DBConnection.run_transaction/4
       (paper_trail) lib/paper_trail.ex:96: PaperTrail.insert!/2
       test/paper_trail/uuid_test.exs:31: (test)
```

```
3) test versioning models that have a non-regular primary key (PaperTrailTest.UUIDTest)
     test/paper_trail/uuid_test.exs:58
     ** (Ecto.ChangeError) value `""` for `PaperTrail.Version.item_id` in `insert` does not match type Ecto.UUID
     code: |> PaperTrail.insert!()
     stacktrace:
       (ecto) lib/ecto/repo/schema.ex:884: Ecto.Repo.Schema.dump_field!/6
       (ecto) lib/ecto/repo/schema.ex:891: anonymous fn/6 in Ecto.Repo.Schema.dump_fields!/5
       (stdlib) maps.erl:257: :maps.fold_1/3
       (ecto) lib/ecto/repo/schema.ex:889: Ecto.Repo.Schema.dump_fields!/5
       (ecto) lib/ecto/repo/schema.ex:824: Ecto.Repo.Schema.dump_changes!/6
       (ecto) lib/ecto/repo/schema.ex:219: anonymous fn/15 in Ecto.Repo.Schema.do_insert/3
       (ecto) lib/ecto/repo/schema.ex:128: Ecto.Repo.Schema.insert!/3
       (paper_trail) lib/paper_trail.ex:111: anonymous fn/3 in PaperTrail.insert!/2
       (ecto_sql) lib/ecto/adapters/sql.ex:784: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
       (db_connection) lib/db_connection.ex:1341: DBConnection.run_transaction/4
       (paper_trail) lib/paper_trail.ex:96: PaperTrail.insert!/2
       test/paper_trail/uuid_test.exs:62: (test)
```

```
10) test [multi tenant] last works (PaperTrailTest.Version)
     test/paper_trail/paper_trail_version_test.exs:128
     ** (Ecto.ChangeError) value `1` for `PaperTrail.Version.item_id` in `insert_all` does not match type Ecto.UUID
     code: add_three_versions(MultiTenant.tenant())
     stacktrace:
       (ecto) lib/ecto/repo/schema.ex:884: Ecto.Repo.Schema.dump_field!/6
       (ecto) lib/ecto/repo/schema.ex:97: anonymous fn/5 in Ecto.Repo.Schema.init_mapper/3
       (elixir) lib/enum.ex:1424: anonymous fn/3 in Enum.map_reduce/3
       (stdlib) maps.erl:257: :maps.fold_1/3
       (elixir) lib/enum.ex:1941: Enum.map_reduce/3
       (ecto) lib/ecto/repo/schema.ex:82: anonymous fn/5 in Ecto.Repo.Schema.extract_header_and_fields/5
       (elixir) lib/enum.ex:1418: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
       (ecto) lib/ecto/repo/schema.ex:44: Ecto.Repo.Schema.do_insert_all/6
       test/paper_trail/paper_trail_version_test.exs:144: PaperTrailTest.Version.add_three_versions/1
       test/paper_trail/paper_trail_version_test.exs:129: (test)
```

I asked my colleague to run those tests again in both branches and they passed. Right now I'm wondering if there are any configurations or anything specific for new test files, as even adding test with `2 == 2` assertion made aforementioned tests fail.

**Comment by izelnakri** [comment]
Hi @haylen , its been 2 years since this PR is open and the changes here are out of sync with the latest version. Feel free to open a new branch & PR if you'd like me to review and introduce this functionality, in the meantime I'm closing the PR.  Thanks for the work so far.

**Pull request closed without merging**